### PR TITLE
Options to enable release and debug assertions

### PR DESCRIPTION
**Pull request by nagisa**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -142,30 +152,46 @@ impl FuzzProject {
...
+            "-Cpasses=sancov",
+            "-Cllvm-args=-sanitizer-coverage-level=3",
+            "-Zsanitizer=address",
+            "-Cpanic=abort",
+        ];
+        flags.extend(other_flags.split(" "));
+        let rustflags = if assertions {
+            &flags[..]
+        } else {
+            &flags[1..]
```
Comment (frewsxcv) [inline src/main.rs]: This slicing logic seems like an easy way for a bug to creep up in the future. Why not just `insert` or `extend` the `Vec` instead?

**Comment by frewsxcv** [comment]
FYI, Travis found an issue here

File: src/main.rs [hunk 2]
```diff
@@ -39,9 +39,19 @@ fn main() {
...
             .setting(AppSettings::TrailingVarArg)
-            .arg(Arg::with_name("TARGET").required(true)
+            .arg(Arg::with_name("release")
+                 .long("release")
+                 .short("O")
+                 .help("Build artifacts in release mode, with optimizations"))
+            .arg(Arg::with_name("debug_assertions")
+                 .long("debug-assertions")
+                 .short("a")
+                 .help("Build artifacts with debug assertions enabled (default if not -O)"))
```
Comment (frewsxcv) [inline src/main.rs]: Is it possible to specify both -a and -O?

Comment (nagisa) [inline src/main.rs]: Sure. It’s an option.

**Comment by Manishearth** [comment]
r+

**Comment by nagisa** [comment]
bors r=Manishearth

**Pull request merged**